code/__HELPERS/icons.dm

```diff
@@ -1370,3 +1370,26 @@ GLOBAL_LIST_EMPTY(transformation_animation_objects)
 	copy.underlays = underlays_to_keep
 
 	return copy
+
+/// Returns the (isolated) security HUD icon for the given job.
+/proc/get_job_hud_icon(datum/job/job) as /icon
+	RETURN_TYPE(/icon)
+	var/static/list/icon_cache = list()
+
+	if(isnull(job))
+		return
+	if(!is_job(job))
+		if(ispath(job, /datum/job))
+			SSjob.get_job_type(job)
+		else if(istext(job))
+			SSjob.get_job(job)
+		if(isnull(job))
+			return null
+
+	//add it to the cache if it isn't already
+	var/job_type = job.type
+	if(!icon_cache[job_type])
+		var/datum/job/job_instance = locate(job_type) in SSjob.all_occupations
```

Comment:
Why are `locate`ing here, we have the job instance already by now (in fact we just accessed it to get `job.type`)